### PR TITLE
Cmake

### PR DESCRIPTION
**Pull request by uwerat**

### CODE REVIEW

File: .clang-format [hunk 1]
```diff
@@ -91,10 +91,12 @@ PenaltyBreakString: 1000
 PenaltyExcessCharacter: 1000000
 PenaltyReturnTypeOnItsOwnLine: 60
 PointerAlignment: Left
-RawStringFormats: 
-  - Delimiter:       pb
-    Language:        TextProto
-    BasedOnStyle:    google
+RawStringFormats:
```
Comment (vrcomputing) [inline .clang-format:94]: @uwerat upgrade

File: CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,118 @@
+############################################################################
+# QSkinny - Copyright (C) 2016 Uwe Rathmann
+# This file may be used under the terms of the QSkinny License, Version 1.0
+############################################################################
+
+cmake_minimum_required(VERSION 3.16.3) # unclear - TODO
```
[the remark this answers is not in the document]
Comment (uwerat) [reply CMakeLists.txt:6]: @rick-vogel

File: cmake/QskBuildFunctions.cmake [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+function(qsk_add_library target)
+
+    if(QT_VERSION_MAJOR VERSION_GREATER_EQUAL 6)
+        qt6_add_library(${ARGV})
+    else()
+        add_library(${ARGV})
+    endif()
+
+    set_target_properties(${target} PROPERTIES
+        LIBRARY_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/lib" )
```
[the remark this answers is not in the document]
Comment (uwerat) [reply cmake/QskBuildFunctions.cmake:28]: done

File: CMakeLists.txt (hunk 2 above)
[the remark this answers is not in the document]
Comment (rick-vogel) [reply CMakeLists.txt:6]: @uwerat sucesfully testet with Qt 5.15 + Qt 6.4.3 + CMake 3.16.3 on Windows

File: .clang-format [hunk 4]
```diff
@@ -112,5 +114,6 @@ SpacesInSquareBrackets: true
 Standard:        Cpp11
 TabWidth:        4
 UseTab:          Never
+UseCRLF:         Always
```
Comment (rick-vogel) [inline .clang-format:117]: Doesn't work with clang-format 12:
```
c:\repositories\qskinny_vrcomputing\.clang-format:117:18: error: invalid boolean
UseCRLF:         Always
                 ^~~~~~
Error reading c:\repositories\qskinny_vrcomputing\.clang-format: invalid argument
```

(deprecated) see: https://clang.llvm.org/docs/ClangFormatStyleOptions.html#usecrlf

upgrade to clang-format 16's

see: https://clang.llvm.org/docs/ClangFormatStyleOptions.html#lineending

alternatively 

```
UseCRLF:         true
```

**Pull request closed without merging**